source/ada/lsp-ada_contexts.adb

```diff
@@ -196,6 +211,18 @@ package body LSP.Ada_Contexts is
       Object.Initialize (Self.LAL_Context, Item);
       Self.Documents.Insert (Item.uri, Object);
 
+      declare
+         Name : constant LSP.Types.LSP_String := Self.URI_To_File (Item.uri);
+         File : constant Virtual_File := Create
+           (Filesystem_String (LSP.Types.To_UTF_8_String (Name)),
+            Normalize => True);
+      begin
+         if not Self.Is_Part_Of_Project (File) then
```

Comment:
Please add a comment explaining why we are doing this. That might be obvious now but not 2, or 3 years later